FastPriorityQueue.js

```diff
@@ -287,13 +287,22 @@ FastPriorityQueue.prototype.forEach = function(callback) {
 FastPriorityQueue.prototype.kSmallest = function(k) {
   if (this.size == 0) return [];
   k = Math.min(this.size, k);
-  var fpq = new FastPriorityQueue(this.compare);
-  const newSize = Math.min((k > 0 ? Math.pow(2, k - 1) : 0) + 1, this.size);
+  const newSize = Math.min(this.size, (1 << (k - 1)) + 1);
```

Comment:
We had a guard against k == 0 which this PR drops. When k = 0, we get (1<<(-1))+1 which is a silly value.

Reply:
If we guard against k == 0 previously, then this code will look safer (k==1 is fine, k < 1 is not).

Reply:
Updated with the guard you suggested below.